src/client.ts

```diff
@@ -43,39 +32,88 @@ export type AcmeGroup = Opaque<any, 'AcmeGroup'>;
 export class APIClient {
   constructor(readonly config: IntegrationConfig) {}
 
-  public async verifyAuthentication(): Promise<void> {
-    // TODO make the most light-weight request possible to validate
-    // authentication works with the provided credentials, throw an err if
-    // authentication fails
-    const request = new Promise<void>((resolve, reject) => {
-      http.get(
-        {
-          hostname: 'localhost',
-          port: 443,
-          path: '/api/v1/some/endpoint?limit=1',
-          agent: false,
-          timeout: 10,
-        },
-        (res) => {
-          if (res.statusCode !== 200) {
-            reject(new Error('Provider authentication failed'));
-          } else {
-            resolve();
-          }
-        },
-      );
+  private accessToken: string;
```

Comment:
TEMP: Leaving it as a non static variable for the meantime because we are getting a new `accessToken` per request.

https://github.com/JupiterOne/graph-checkmarx/blob/5301833e04eca44cf0df303d68a79525bd4e549a/src/client.ts#L33

The problem I'm having with this approach is after the integration steps have been done, the generated `accessToken` remains valid. It feels like it might be a security concern. I will make it a `static` class if you guys are okay with leaving it as is or if we find a way to run a `logout()` call after integration.

Reply:
Non-static is fine :+1: , it'd just mean that each step will get a new cookie (e.g. each step instantiates a new client and it gets a new cookie), so long the Cybereason doesn't mind a single account requesting many cookies at "once", it should work. Let's try it and see.

P.S I'd still name with some name that has the word 'cookie' in it, the value in the example you were referencing was indeed the access_token, hence that name :slightly_smiling_face: